src/instana/span/registered_span.py

```diff
@@ -1,11 +1,21 @@
 # (c) Copyright IBM Corp. 2024
 
+from typing import TYPE_CHECKING
+
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.trace import SpanKind
+
 from instana.log import logger
 from instana.span.base_span import BaseSpan
-from instana.span.kind import ENTRY_SPANS, EXIT_SPANS, HTTP_SPANS, LOCAL_SPANS
+from instana.span.kind import (
+    ENTRY_SPANS,
+    EXIT_SPANS,
+    HTTP_SPANS,
+    LOCAL_SPANS,
+)
 
-from opentelemetry.trace import SpanKind
-from opentelemetry.semconv.trace import SpanAttributes
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 
 class RegisteredSpan(BaseSpan):
```

Comment:
Could we add typing hints to the functions we worked on? All missing typing hints will be covered as we create PRs.
The parameters of the `__init__` function could be updated accordingly for this PR.